### ISSUE TITLE
Replacing static import with dynamic import causes spurious warnings

### ISSUE THREAD

**Issue by Kovensky** [Microsoft/TypeScript#16291]
**TypeScript Version:**  nightly (2.4.0-dev.20170606)

**Code**

```ts
declare global {
  interface System {
    import<T = any> (module: string): Promise<T>
  }

  declare var System: System
}

import * as fooNs from './foo'

System.import<typeof fooNs>('./foo')
.then(foo => {
  // do things with foo
})
```

Given the above code, replacing `System.import<typeof fooNs>(` with `import(`, and then deleting the static import causes a "module not found" error in the IDE service.

The error goes away if other changes are made to the file and it is saved again.

**Expected behavior:**

No error.

**Actual behavior:**

Transient error.

**Comment by yuit** [comment]
Thanks @Kovensky  for giving this feature a shot

**Comment by mikehaas763** [comment]
@Kovensky Is this issue related to supporting the `import()` syntax in Typescript? I'm looking for the same thing but haven't been able to find much info.

**Comment by yuit** [comment]
@mhegazy  helps me repo the issue.  We have repo and issue the fix. It will be in nightly

@mikehaas763  this issue is related to supporting `import()`... It will be in 2.4 RC
